ports/rp2/machine_adc.c

```diff
@@ -72,18 +82,26 @@ STATIC mp_obj_t machine_adc_make_new(const mp_obj_type_t *type, size_t n_args, s
 
     } else {
         // Get GPIO and check it has ADC capabilities.
-        channel = mp_hal_get_pin_obj(source);
-        if (!ADC_IS_VALID_GPIO(channel)) {
+        const machine_pin_obj_t *pin = machine_pin_find(source);
+        channel = pin->id;
+        #if MICROPY_HW_ADC_EXT_COUNT
+        is_ext = pin->is_ext;
+        #endif
+        if (!ADC_IS_VALID_GPIO(channel) && !machine_pin_ext_is_adc(channel)) {
```

Comment:
This should be:
```c
if (!ADC_IS_VALID_GPIO(channel) && !(is_ext && machine_pin_ext_is_adc(channel))) {
```
But maybe it's clearer to write it in the reverse logic:
```c
if (!(ADC_IS_VALID_GPIO(channel) || (is_ext && machine_pin_ext_is_adc(channel)))) {
```

Reply:
It should be `!` 
```
if (!ADC_IS_VALID_GPIO(channel) && !(is_ext && machine_pin_ext_is_adc(channel))) {
```

Reply:
Note this doesn't look right either
```c
if (!(ADC_IS_VALID_GPIO(channel) || (is_ext && !machine_pin_ext_is_adc(channel)))) {
```
If it's not an valid internal GPIO and not external pin it will not raise an exception